common/authorization/defaultClaimMapper.go

```diff
@@ -0,0 +1,121 @@
+// The MIT License
+//
+// Copyright (c) 2020 Temporal Technologies Inc.  All rights reserved.
+//
+// Copyright (c) 2020 Uber Technologies, Inc.
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+
+package authorization
+
+import (
+	"fmt"
+	"strings"
+
+	"github.com/dgrijalva/jwt-go"
+)
+
+// Default claim mapper that gives system level admin permission to everybody
+type defaultClaimMapper struct {
+	keyProvider TokenKeyProvider
+}
+
+func NewDefaultClaimMapper(provider TokenKeyProvider) ClaimMapper {
+	return &defaultClaimMapper{keyProvider: provider}
+}
+
+var _ ClaimMapper = (*defaultClaimMapper)(nil)
+
+func (a *defaultClaimMapper) GetClaims(authInfo AuthInfo) (*Claims, error) {
+
+	claims := Claims{}
+
+	if authInfo.authToken != "" {
+		parts := strings.Split(authInfo.authToken, " ")
+		if len(parts) != 2 {
+			return nil, fmt.Errorf("unexpected authorization token format")
+		}
+		name := strings.ToLower(parts[0])
+		if name != "bearer" {
+			return nil, fmt.Errorf("unexpected name in authorization token: %s", name)
+		}
+		jwtClaims, err := parseJWT(parts[1], a.keyProvider)
+		if err != nil {
+			return nil, err
+		}
+		claims.subject = jwtClaims["sub"].(string)
```

Comment:
[Nit] when doing this type assertion, this will panic if jwtClaims is (for whatever reason) not a string. https://tour.golang.org/methods/15

Might be preferable to use the form that returns a bool and error out gracefully, unless we are 100% sure this would never happen

Reply:
Fix this and similar cases to protect from the unlikely but maybe not impossible cases.